test/CMakeLists.txt

```diff
@@ -9,21 +9,22 @@ add_definitions (-DBINDIR=\"${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/\")
 # Define cmake configuration flags to configure and build external projects with the same flags as specified for
 # this project.
 set (SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "")
+list (APPEND SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}")
```

Comment:
If we do not propagate the C Compiler (set via `CC` env var in CI), googletest will be build with the default C compiler.

Example:
tests are built with `g++-7`, but googletest is built with `gcc-9`.